### ISSUE TITLE
Add modpack dialog

### ISSUE THREAD

**Issue by 7osCraft** [SKCraft/Launcher#63]
hello,
can you do somthing like add modpack dialog to add private modpacks by code Instead of game key in the settings?

**Comment by Collussus** [comment]
If this was implemented, i'd prefer both. This way you can give certain people instant access to all hidden packs, and give others access to singular hidden packs.

**Comment by 7osCraft** [comment]
Ok I prefer both too but change the game key field to password field instead of normal field

**Comment by sk89q** [comment]
Will consider it.

**Comment by sk89q** [comment]
Note that one issue with the modpack-per-URL is that the launcher has to hit all the URLs to check for updates.

On v3 of the launcher, this feature existed, but it only checked for update when you launched.

**Comment by 7osCraft** [comment]
Ok but when will be that implemented?

**Comment by Slind14** [comment]
> Will consider it.

**Comment by sk89q** [comment]
One thing I could do is something that actually adds values to the game key field, which means that looking for updates requires only hitting that one packages list file.

**Comment by 7osCraft** [comment]
Ok do what you see better
sorry for my english I'm arabic

**Comment by sk89q** [comment]
No, I'm asking because I don't need it for myself.

Is it fine if the add modpack button (via URL) secretly just changes the game key field?

**Comment by 7osCraft** [comment]
No, I'want somthing like a dialog to enter a secret code that adds one modpack without changing the game key field

**Comment by Collussus** [comment]
@sk89q 

Rather than checking each code, you only do the one check when you add said code, and from there on out just do update checks as you normally do.

You can assume that the code stays the same regardless of whether someone has changed the mod list.

**Comment by sk89q** [comment]
The update check is used for figuring out whether there needs to be modpack updates. Without that, the instance would never update.

Since login isn't done in the background, perhaps it's fine if I just toss that check during the login process for URL-added packs.

**Comment by Collussus** [comment]
I see what you mean.

I do need to point out though that i'd find it highly unlikely if it's taxing to any server if you hit multiple package lists. If it does tax dropbox hosted services, you could just make it an optional feature.

Anyone on a dedicated server can easily take the extra traffic.

**Comment by sk89q** [comment]
It's not the server I'm worried about -- just the client.

**Comment by Collussus** [comment]
Fair point. Though rather than multiple requests, can't you do a json get request on the packages.php, and get all the relevant information from the single file?

**Comment by sk89q** [comment]
That's sort of what the game key does, so I was asking if that would be a acceptable way of going about this.

The downside with it is that you can't have modpacks hosted at different places with no relationship to the main package list file, and it also requires scripting.

**Comment by 7osCraft** [comment]
Yeah first I made a php file that gets all public packs from anothe packages.json and adds another private package but the problem is that if I changed the game key field it will still get the public modpacks but removes the first private pack and adds anothe one

**Comment by Collussus** [comment]
@sk89q 

Personally i wouldn't mind if users do multiple requests. I'm expecting people to have more than enough CPU & RAM to be able to handle some extra requests. Same for the amount of bandwidth they have.

They're playing minecraft after all ;)

**Comment by sk89q** [comment]
The requests aren't affected by CPU & RAM -- they are affected by your physical distance to the server.

**Comment by 7osCraft** [comment]
what will you do?

**Comment by sk89q** [comment]
I won't know until I get to it. I have [a lot of projects](https://github.com/sk89q).
